acquisitions/src/main/resources/thunderjet/mod-finance/features/batch-transaction-api.feature

```diff
@@ -0,0 +1,217 @@
+    * def invoiceLineId2 = callonce uuid9
+    * def encumbranceId1 = callonce uuid10
+    * def encumbranceId2 = callonce uuid11
+    * def encumbranceId3 = callonce uuid12
+    * def allocationId = callonce uuid13
+    * def pendingPaymentId1 = callonce uuid14
+    * def pendingPaymentId2 = callonce uuid15
+
+
+  Scenario: Create fund and budget
+    * def v = call createFund { id: '#(fundId)' }
+    * def v = call createBudget { id: '#(budgetId)', allocated: 100, fundId: '#(fundId)', status: 'Active' }
+
+
+  Scenario: Create initial transactions without using the batch API
+    Given path 'finance/order-transaction-summaries'
+    And request
+    """
+      {
+        "id": '#(orderId)',
+        "numTransactions": 3
+      }
+
+    """
+    When method POST
+    Then status 201
+
+    Given path 'finance/encumbrances'
+    And request
+    """
+      {
+        "id": "#(encumbranceId1)",
+        "amount": 10,
+        "currency": "USD",
+        "description": "Encumbrance 1",
+        "fiscalYearId": "#(globalFiscalYearId)",
+        "source": "User",
+        "fromFundId": "#(fundId)",
+        "transactionType": "Encumbrance",
+        "encumbrance" : {
+          "initialAmountEncumbered": 10,
+          "status": "Unreleased",
+          "orderType": "One-Time",
+          "subscription": false,
+          "reEncumber": false,
+          "sourcePurchaseOrderId": "#(orderId)",
+          "sourcePoLineId": "#(poLineId1)"
+        }
+      }
+    """
+    When method POST
+    Then status 201
+
+    Given path 'finance/encumbrances'
+    And request
+    """
+      {
+        "id": "#(encumbranceId2)",
+        "amount": 5,
+        "currency": "USD",
+        "description": "Encumbrance 2",
+        "fiscalYearId": "#(globalFiscalYearId)",
+        "source": "User",
+        "fromFundId": "#(fundId)",
+        "transactionType": "Encumbrance",
+        "encumbrance" : {
+          "initialAmountEncumbered": 5,
+          "status": "Unreleased",
+          "orderType": "One-Time",
+          "subscription": false,
+          "reEncumber": false,
+          "sourcePurchaseOrderId": "#(orderId)",
+          "sourcePoLineId": "#(poLineId2)"
+        }
+      }
+    """
+    When method POST
+    Then status 201
+
+    Given path 'finance/encumbrances'
+    And request
+    """
+      {
+        "id": "#(encumbranceId3)",
+        "amount": 3,
+        "currency": "USD",
+        "description": "Encumbrance 3",
+        "fiscalYearId": "#(globalFiscalYearId)",
+        "source": "User",
+        "fromFundId": "#(fundId)",
+        "transactionType": "Encumbrance",
+        "encumbrance" : {
+          "initialAmountEncumbered": 3,
+          "status": "Unreleased",
+          "orderType": "One-Time",
+          "subscription": false,
+          "reEncumber": false,
+          "sourcePurchaseOrderId": "#(orderId)",
+          "sourcePoLineId": "#(poLineId3)"
+        }
+      }
+    """
+    When method POST
+    Then status 201
```

Comment:
we can use scenario outline: to avoid duplicaion

Reply:
good idea